sqlmesh/core/engine_adapter/athena.py

```diff
@@ -0,0 +1,441 @@
+    try:
+        from pyathena.error import OperationalError  # type: ignore
+
+        return retry(
+            retry=retry_if_exception_type(OperationalError),
+            stop=stop_after_attempt(3),
+            wait=wait_fixed(5),
+            reraise=True,
+        )(func)
+    except ImportError:
+        # if pyathena isnt installed, this is a no-op
+        return func
+
+
+class AthenaEngineAdapter(PandasNativeFetchDFSupportMixin):
+    DIALECT = "athena"
+    SUPPORTS_TRANSACTIONS = False
+    SUPPORTS_REPLACE_TABLE = False
+    # Athena has the concept of catalogs but no notion of current_catalog or setting the current catalog
+    CATALOG_SUPPORT = CatalogSupport.UNSUPPORTED
+    # Athena's support for table and column comments is too patchy to consider "supported"
+    # Hive tables: Table + Column comments are supported
+    # Iceberg tables: Column comments only
+    # CTAS, Views: No comment support at all
+    COMMENT_CREATION_TABLE = CommentCreationTable.UNSUPPORTED
+    COMMENT_CREATION_VIEW = CommentCreationView.UNSUPPORTED
+    SCHEMA_DIFFER = TrinoEngineAdapter.SCHEMA_DIFFER
+
+    def __init__(
+        self, *args: t.Any, s3_warehouse_location: t.Optional[str] = None, **kwargs: t.Any
+    ):
+        # Need to pass s3_warehouse_location to the superclass so that it goes into _extra_config
+        # which means that EngineAdapter.with_log_level() keeps this property when it makes a clone
+        super().__init__(*args, s3_warehouse_location=s3_warehouse_location, **kwargs)
+        self.s3_warehouse_location = s3_warehouse_location
+
+    @property
+    def s3_warehouse_location(self) -> t.Optional[str]:
+        return self._s3_warehouse_location
+
+    @s3_warehouse_location.setter
+    def s3_warehouse_location(self, value: t.Optional[str]) -> None:
+        if value:
+            value = _ensure_valid_location(value)
+        self._s3_warehouse_location = value
+
+    def create_state_table(
+        self,
+        table_name: str,
+        columns_to_types: t.Dict[str, exp.DataType],
+        primary_key: t.Optional[t.Tuple[str, ...]] = None,
+    ) -> None:
+        self.create_table(
+            table_name,
+            columns_to_types,
+            primary_key=primary_key,
+            table_properties={
+                # it's painfully slow, but it works
+                "table_type": exp.Literal.string("iceberg")
+            },
+        )
+
+    def _get_data_objects(
+        self, schema_name: SchemaName, object_names: t.Optional[t.Set[str]] = None
+    ) -> t.List[DataObject]:
+        """
+        Returns all the data objects that exist in the given schema and optionally catalog.
+        """
+        schema_name = to_schema(schema_name)
+        schema = schema_name.db
+        query = (
+            exp.select(
+                exp.case()
+                .when(
+                    # 'awsdatacatalog' is the default catalog that is invisible for all intents and purposes
```

Comment:
This doesn't quite explain why do we set catalog to NULL of the actual value is `awsdatacatalog`. What does "invisible" actually mean here?

Reply:
It's because the integration test `test_get_data_objects` expects that if it passes a schema like `test_schema_x` (as opposed to a catalog-schema combo like `test_catalog.test_schema_x`) to `get_data_objects()`, the resulting data objects should have None set on the catalog property.

I'll amend the comment